src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -716,7 +739,16 @@ private void exportOutpatient(Person person, long stopTime)
           fieldValues.put(OUTPATIENT.PRNCPAL_DGNS_CD, mappedDiagnosisCodes.get(0));
         }
       }
-
+      // Check for external code...
+      String primary = fieldValues.get(OUTPATIENT.PRNCPAL_DGNS_CD);
```

Comment:
This code block is repeated for each of the claim files with external codes. Can you DRY it out into a single method that takes fieldValues, source field enum value, external field enum value and version field enum value?

Reply:
It also needs to take the person (as a source of randomness), and in one case the presentOnAdmission set and presentOnAdmission enum value. So that is seven arguments, which is kind of a lot, but I'll give it a try.